flink-ml-lib/src/main/java/org/apache/flink/ml/feature/minmaxscaler/MinMaxScaler.java

```diff
@@ -0,0 +1,210 @@
+                            public void mapPartition(
+                                    Iterable<DenseVector> values,
+                                    Collector<MinMaxScalerModelData> out) {
+                                Iterator<DenseVector> iter = values.iterator();
+                                DenseVector minVector = iter.next();
+                                DenseVector maxVector = iter.next();
+                                out.collect(new MinMaxScalerModelData(minVector, maxVector));
+                            }
+                        });
+
+        MinMaxScalerModel model =
+                new MinMaxScalerModel().setModelData(tEnv.fromDataStream(modelData));
+        ReadWriteUtils.updateExistingParams(model, getParamMap());
+        return model;
+    }
+
+    /**
+     * A stream operator to compute the min and max values in each partition of the input bounded
+     * data stream.
+     */
+    private static class MinMaxReduceFunctionOperator extends AbstractStreamOperator<DenseVector>
+            implements OneInputStreamOperator<DenseVector, DenseVector>, BoundedOneInput {
+        private ListState<DenseVector> minState;
+        private ListState<DenseVector> maxState;
+
+        private DenseVector minVector;
+        private DenseVector maxVector;
+
+        @Override
+        public void endInput() {
+            if (minVector != null) {
+                output.collect(new StreamRecord<>(minVector));
+                output.collect(new StreamRecord<>(maxVector));
+            }
+        }
+
+        @Override
+        public void processElement(StreamRecord<DenseVector> streamRecord) {
+            DenseVector currentValue = streamRecord.getValue();
+            if (minVector == null) {
+                int vecSize = currentValue.size();
+                minVector = new DenseVector(vecSize);
+                maxVector = new DenseVector(vecSize);
+                System.arraycopy(currentValue.values, 0, minVector.values, 0, vecSize);
+                System.arraycopy(currentValue.values, 0, maxVector.values, 0, vecSize);
+            } else {
+                Preconditions.checkArgument(
+                        currentValue.size() == maxVector.size(),
+                        "CurrentValue should has same size with maxVector.");
+                for (int i = 0; i < currentValue.size(); ++i) {
+                    minVector.values[i] = Math.min(minVector.values[i], currentValue.values[i]);
+                    maxVector.values[i] = Math.max(maxVector.values[i], currentValue.values[i]);
+                }
+            }
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public void initializeState(StateInitializationContext context) throws Exception {
+            super.initializeState(context);
+            minState =
+                    context.getOperatorStateStore()
+                            .getListState(
+                                    new ListStateDescriptor<>(
```

Comment:
Hmm.. instead of using `getOperatorConfig().getTypeSerializerIn(...)`, would it be better to use the code below for simplicity and consistency with other usages of `getListState(...)` in Flink ML?

```
new ListStateDescriptor<>("minState", TypeInformation.of(DenseVector.class))
```